### DISCUSSION TITLE
SX1278 Output Power change.

### DISCUSSION THREAD

**Discussion by ghost** [jgromes/RadioLib#308] [Q&A]
Hello, I'm trying to use RFM98PW (30dBm 433MHz SX1278) module with this library. I'm trying to send data from a range of 8km. The SX1278 Transmit code works just fine but I could only achieve a 3.2km range so far.  Upon reading the library for SX1278 I realised the output power is limited by 17dBm and the input current is limited by 100mA. I need help rearranging the limitations for this module. 

I need the output power to be 30dBm and the max input current to be 700mA.

I thought of changing the values where it's checking whether the power is lower than 17dBm for example as well as the current limit value but I don't want to burn the module accidentally. Also I don't know whether or not I need to change anything with the gain value. Any help is appreciated.

PS: I'm using RadioLib version 4.4.2
![image](https://user-images.githubusercontent.com/58036865/120905394-7d829c00-c65a-11eb-862b-232f0a9cf023.png)

**Comment by jgromes** [comment]
Same as other high-power LoRa modules, e.g https://github.com/jgromes/RadioLib/issues/184

SX1278 on its own can only output 20 dBm, that's what the silicon was designed for. To get higher output power, there's an external amplifier inside the RFM98 module, which amplifies output from the SX1278. Therefore, to set the correct power output, you need to know what is the gain of the external amplifier. Briefly going through RFM98PW docs shows nothing, so I suggest contacting HopeRF directly, or removing the shield from the module and checking the IC for yourself (bit risky, as you'll have to reattach the shield afterwards).

Similarly with the current limit, it's just the input current for SX1278 alone, without the external PA. Therefore, you don't have to change it as the input current into the P/A won't affect the input current into SX1278.

Let me know if you figure out which PA they're using, I've seen a couple different options and I'm curious what they went with.

**Reply by ghost** [reply]
The module has a max power output of 30dBm (1W) as I mentioned and SX1278 outputs at 20dBm (100mW). I will still contact HopeRF but isn't the gain 10? 1000mW/100mW = 10 or 30-20 = 10dB. If I know the gain value how do I change it in the library?

**Reply by jgromes** [reply]
SX1278 power output is configurable, the maximum is +20 dBm, but that tells you nothing about the external PA gain. For example, on Ebyte E22-900M30S which uses a similar setup, there's a Y2233W PA with a +26 dB gain, so you have to set SX1268 output to 4 dBm to get 30 dBm of total output.

If you know the gain of the PA, you set the output from SX1278 as in the example above to hit that 30 dBm target. If you set the SX1278 output too high, you will saturate the PA and distort the output signal. If you set the output too low, it won't be enough to drive the PA. Also, with some cheaper PAs, the gain is frequency-dependent, so it's critical to know which exact PA is in there.
